lib/linter-rust.coffee

```diff
@@ -2,47 +2,77 @@ linterPath = atom.packages.getLoadedPackage("linter").path
 Linter = require "#{linterPath}/lib/linter"
 
 {exec} = require 'child_process'
-{log, warn} = require "#{linterPath}/lib/utils"
+{log, warn, findFile} = require "#{linterPath}/lib/utils"
 path = require 'path'
 
 
 class LinterRust extends Linter
   @enable: false
   @syntax: 'source.rust'
+  @cargoPath: 'cargo'
+  @cargoManifestPath: null
   linterName: 'rust'
   errorStream: 'stderr'
-  regex: '^(.+):(?<line>\\d+):(?<col>\\d+):\\s*(\\d+):(\\d+)\\s+((?<error>error|fatal error)|(?<warning>warning)):\\s+(?<message>.+)\n'
+  regex: '^(?<file>.+):(?<line>\\d+):(?<col>\\d+):\\s*(\\d+):(\\d+)\\s+((?<error>error|fatal error)|(?<warning>warning)):\\s+(?<message>.+)\n'
 
   constructor: (@editor) ->
     super @editor
     atom.config.observe 'linter-rust.executablePath', =>
       @executablePath = atom.config.get 'linter-rust.executablePath'
       exec "#{@executablePath} --version", @executionCheckHandler
+    atom.config.observe 'linter-rust.executablePath2', =>
+      @cargoPath = atom.config.get 'linter-rust.executablePath2'
 
   executionCheckHandler: (error, stdout, stderr) =>
-    versionRegEx = /rustc ([\d\.]+)/
+    versionRegEx = /(rustc|cargo) ([\d\.]+)/
     if not versionRegEx.test(stdout)
       result = if error? then '#' + error.code + ': ' else ''
       result += 'stdout: ' + stdout if stdout.length > 0
       result += 'stderr: ' + stderr if stderr.length > 0
-      console.error "Linter-Rust: \"#{@executablePath}\" was not executable: \
+      console.error "Linter-Rust: \"#{@executablePath}\" was invalid: \
       \"#{result}\". Please, check executable path in the linter settings."
     else
       @enabled = true
-      log "Linter-Rust: found rust " + versionRegEx.exec(stdout)[1]
-      do @initCmd
+      log "Linter-Rust: found " + stdout
+      log 'Linter-Rust: initialization completed'
 
-  initCmd: =>
-    @cmd = "#{@executablePath} --no-trans --color never"
-    log 'Linter-Rust: initialization completed'
+  initCmd: (editing_file) =>
+    # search for Cargo.toml in container directoies
+    dir = path.dirname editing_file
+    @cargoManifestPath = findFile(dir, "Cargo.toml")
+    if @cargoManifestPath
+      log "found Cargo.toml: #{@cargoManifestPath}"
+      @cmd = "cargo build"
+      @cwd = path.dirname @cargoManifestPath
+    else
+      @cmd = "rustc --no-trans --color never"
```

Comment:
rustc --no-trans has been depreciated as of  [#19900](https://github.com/rust-lang/rust/pull/19900). You should use ```-Z no-trans``` instead.